example/extension/sqlinject/demo/sqlinject.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	"os"
+	"time"
+)
+
+func main() {
+	var mysqlDSN string
+	mysqlDSN = os.Getenv("MYSQL_DSN")
+	if mysqlDSN == "" {
+		mysqlDSN = "test:test@tcp(127.0.0.1:3306)/test"
+	}
+	db, err := sql.Open("mysql", mysqlDSN)
+	if err != nil {
+		return
+	}
+	// See "Important settings" section.
+	db.SetConnMaxLifetime(time.Minute * 3)
+	db.SetMaxOpenConns(10)
+	db.SetMaxIdleConns(10)
+
+	if _, err := db.ExecContext(context.Background(), `CREATE TABLE IF NOT EXISTS usersx (id char(255), name VARCHAR(255), age INTEGER)`); err != nil {
+		fmt.Printf("exec create error: %v", err)
+	}
+
+	if _, err := db.ExecContext(context.Background(), `INSERT INTO usersx (id, name, age) VALUE ( ?, ?, ?)`, "0", "foo", 10); err != nil {
+		fmt.Printf("exec insert error: %v", err)
+	}
+
+	// test sql inject
+	if _, err := db.Query("SELECT * FROM userx WHERE id = '0' AND name = 'foo'"); err != nil {
```

Comment:
One nit: it seems there is no sql injection demonstrated, maybe something like:
```
db.Query("SELECT * FROM userx WHERE id = '0' AND name = %s", "''10' AND 1=1");
```

Reply:
> One nit: it seems there is no sql injection demonstrated, maybe something like:
> 
> ```
> db.Query("SELECT * FROM userx WHERE id = '0' AND name = %s", "''10' AND 1=1");
> ```

I will polish the example, use `+` or `StringBuilder` to build the sql is better to demonstrate the sql injection.

Reply:
> One nit: it seems there is no sql injection demonstrated, maybe something like:
> 
> ```
> db.Query("SELECT * FROM userx WHERE id = '0' AND name = %s", "''10' AND 1=1");
> ```

I ’ve polished it with a demo that concatenates sql with multiple string in #169